src/components/Icon/Icon.tsx

```diff
@@ -21,6 +22,29 @@ interface Props extends IconProps {}
 
 export function Icon({source, color, backdrop, accessibilityLabel}: Props) {
   const i18n = useI18n();
+  const unstableMonorail = useMonorail();
+
+  /* eslint-disable babel/camelcase */
+  useEffect(() => {
+    if (unstableMonorail) {
+      let loggedSource;
+      if (typeof source === 'function') {
+        loggedSource = source.name;
+      } else if (source === 'placeholder') {
+        loggedSource = source;
+      } else {
+        loggedSource = 'custom icon string';
```

Comment:
@kaelig this is how I decided to log custom string icons, what do you think? I can't log a filename like I do for the components because this is just a string

Reply:
Thanks, that's probably the right compromise to not impact performance.

Is there any additional info we could log that would give us a hint on what/where this is rendered?

Reply:
I thought the same thing, it'd be nice to know where this icon is being rendered, I don't think so because the code gets minified. I have to try it out running in actual web though, an example in the playground wouldn't be enough here

Reply:
I couldn't figure out how to do this in web, I wasn't able to run web with production assets to be able to test :/